### PR TITLE
Added inputs varsub & validation, and --input flag

### LINKED ISSUE #97: Support for input variables in varsub
Input variables are currently completely ignored in varsub. They're parsed, but they're not used.

There's three parts to this:

1. The default values from the input variables definitions in the `.wharf-ci.yml` file need to be added to the `varsub.SourceSlice` in `pkg/wharfyml/definition.go`, before the varsub substitution is applied on the environment variables. The default values should have lowest priority in the varsub, so they should be added AFTER the `varsub.Source` provided by the `wharfyml.Args`

2. User should be able to feed in input variables through `wharfyml.Args`. Setting input variables that doesn't exist in the `.wharf-ci.yml` file should raise an error. These values added through `wharfyml.Args` should also be added to `varsub.SourceSlice`, and with higher priority so that it can override the default values provided by the input variables definitions **as well as the `varsub.Source` provided by the `wharfyml.Args`**.

3. Add flag to `wharf run` and `wharf vars` to allow setting variables via the CLI. Such as:

   ```console
   $ wharf run --input my-input-var=foobar --input my-other-var=moodoo
   $ wharf run -i my-input-var=foobar -i my-other-var=moodoo
   ```

## Skip for now

- Adding completions for the input variable flag

### PR DESCRIPTION
**Pull request by jilleJr**
- \[x] I've added a new note in the `CHANGELOG.md` file, according to docs:
  https://iver-wharf.github.io/#/development/changelogs/writing-changelogs

## Summary

- Added input vars varsub
- Added input vars sample in `test/kubectl`
- Added `wharfyml.Args.Inputs` with validation parsing
- Added `--input` flag

## Motivation

Closes #97

## Preview

```yaml
# .wharf-ci.yml
inputs:
  - name: file
    type: string
    default: somefile.txt

test:
  step1:
    container:
      image: alpine:latest
      cmds:
      - echo 'Hello world!'
      - echo 'output from ${file}:'
      - cat ${file}
  step2:
    container:
      image: busybox:latest
      cmds:
      - date
```

```console
$ wharf vars
[INFO ] Found 13 variables from 3 different sources:
	
	.wharf-ci.yml, input defaults:
	  file  somefile.txt

	//snip

$ wharf vars -i file=otherfile.txt --all
[INFO ] Found 13 variables from 3 different sources:
	
	.wharf-ci.yml, input defaults:
	  file  somefile.txt

	//snip

	.wharf-ci.yml, overridden input values:
	  file  otherfile.txt

$ wharf run
[INFO |TARSTORE] Copying files.  src=/home/kalle/dev/wharf/wharf-cmd/test/container  dst=/tmp/wharf-cmd-repo-3019035509/full
[INFO |TARSTORE] Creating tarball.  path=/tmp/wharf-cmd-repo-3019035509/full.tar
[INFO ] Starting build.
[INFO ] Starting stage.  stages=0/1  stage=test
[INFO ] Starting step.  steps=0/2  stage=test  step=step2
[INFO ] Starting step.  steps=0/2  stage=test  step=step1
[INFO |test/step2] 2022-04-27T10:05:04.282797659Z Wed Apr 27 10:05:04 UTC 2022
[INFO ] Done with step.  steps=1/2  stage=test  step=step2  dur=8s
[INFO |test/step1] 2022-04-27T10:05:05.932771337Z Hello world!
[INFO |test/step1] 2022-04-27T10:05:05.932814950Z output from somefile.txt:
[INFO |test/step1] 2022-04-27T10:05:05.933238235Z Hello from somefile.txt
[INFO ] Done with step.  steps=2/2  stage=test  step=step1  dur=10s
[INFO ] Done with stage.  stages=1/1  stage=test  dur=10s
[INFO ] Done with build.  dur=10s  status=Success

$ wharf run -i file=otherfile.txt
[INFO |TARSTORE] Copying files.  src=/home/kalle/dev/wharf/wharf-cmd/test/container  dst=/tmp/wharf-cmd-repo-697301188/full
[INFO |TARSTORE] Creating tarball.  path=/tmp/wharf-cmd-repo-697301188/full.tar
[INFO ] Starting build.
[INFO ] Starting stage.  stages=0/1  stage=test
[INFO ] Starting step.  steps=0/2  stage=test  step=step2
[INFO ] Starting step.  steps=0/2  stage=test  step=step1
[INFO |test/step2] 2022-04-27T10:04:32.279107105Z Wed Apr 27 10:04:32 UTC 2022
[INFO ] Done with step.  steps=1/2  stage=test  step=step2  dur=8s
[INFO |test/step1] 2022-04-27T10:04:33.918700815Z Hello world!
[INFO |test/step1] 2022-04-27T10:04:33.918734544Z output from otherfile.txt:
[INFO |test/step1] 2022-04-27T10:04:33.919366806Z This is from the other file
[INFO ] Done with step.  steps=2/2  stage=test  step=step1  dur=10s
[INFO ] Done with stage.  stages=1/1  stage=test  dur=10s
[INFO ] Done with build.  dur=10s  status=Success
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -55,6 +55,13 @@ This project tries to follow [SemVer 2.0.0](https://semver.org/).
 - Added support for `.gitignore` ignored files and directories when transferring
   repo in `wharf run`. Can be disabled via new `--no-gitignore` flag. (#85)
 
+- Added input variables support, and with `--input, -i` flag to `wharf run` and
+  `wharf vars` commands that allows setting values, ex: (#97)
```
Comment (Alexamakans) [inline CHANGELOG.md:59]: nit: Nice if it was a bit more descriptive.

Something like:
```suggestion
- Added input variables support using the `inputs` field in `.wharf-ci.yml`
  files, and the `--input, -i` flag to `wharf run` and `wharf vars` commands
  through the CLI, ex: (#97)
```
maybe?

File: CHANGELOG.md [hunk 2]
```diff
@@ -55,6 +55,13 @@ This project tries to follow [SemVer 2.0.0](https://semver.org/).
...
+
+  ```sh
+  wharf run --input myInputVar=myValue
+  ```
```
Comment (Alexamakans) [inline CHANGELOG.md:63]: nit: Would be nice if there was an example with multiple input variables. Probably no need for a single value example in that case.

Is it
`--input myInputVar1=myValue1,myInputVar2=myValue2`
or
`--input myInputVar1=myValue1 --input myInputVar2=myValue2`?

And what happens with duplicate keys?

Better place for usage specifics would probably be in docs or something, but at least having >1 values in the example would be nice.

File: pkg/wharfyml/definition.go [hunk 3]
```diff
@@ -48,11 +49,18 @@ func visitDefNode(node *yaml.Node, args Args) (def Definition, errSlice Errors)
 			var errs Errors
 			def.Inputs, errs = visitInputsNode(n.value)
 			errSlice.add(wrapPathErrorSlice(errs, propInputs)...)
+
```
Comment (Alexamakans) [inline pkg/wharfyml/definition.go:52]:
```suggestion
```

**Review by Alexamakans — APPROVED** [review]
LGTM 👍🏻

Only nits, approving

File: CHANGELOG.md (hunk 2 above)
Comment (jilleJr) [reply CHANGELOG.md:64]: On duplicated values the latter will override the former, which IMO is expected.

Good idea with the multiples examples though!

```suggestion
  ```sh
  wharf run --input myInputVar=myValue
  # => [ "myInputVar": "myValue" ]

  # Supports multiple:
  wharf run --input var1=value1 --input var2=value2
  # => [ "var1": "value1", "var2": "value2" ]

  # On collisions, the last value is used:
  wharf run --input myVar=foo --input myVar=bar
  # => [ "myVar": "bar" ]
  ```
```

Comment (jilleJr) [reply CHANGELOG.md]: How about now?

**Pull request merged**